resources/io/jenkins/infra/docker/README.adoc

```diff
@@ -0,0 +1,130 @@
+export CONTAINER_BIN=docker
+# ...
+make lint
+make build
+# ...
+----
+
+* OR prefix make commands with `CONTAINER_BIN=docker`:
+
+[source,bash]
+---
+CONTAINER_BIN=docker make lint
+CONTAINER_BIN=docker make build
+----
+
+== Workflow Steps
+
+The following steps are available:
+
+* Linting with `make lint` to run a static analyzis to ensure good practises are applied the written code.
+
+* Build the image with `make build` to generate a local TAR archive in `image.tar`
+and store it in the local container tool's index under the name defined by `$IMAGE_NAME` (see <<Customization>>).
+** You might want to authenticated your container tool to the Docker Hub to avoid rate limitation for this step.
+
+* Run a test harness with the command `make test`, on the image stored in `image.tar`.
+
+* Cleanup the image and its generated artefacts with `make clean`.
+
+* Execute the "developer workflow" with `make all`, which executes (in this order) `make clean lint build test`.
+
+* Deploy the image with `make deploy` to tag the current image referenced by `$IMAGE_NAME` as `$IMAGE_DEPLOY_NAME`
+(see <<Customization>>) and then push it to the remote repository implied by `$IMAGE_DEPLOY_NAME`.
+** This steps requires the container tool to be authenticated to the remote registry.
+** This step is NOT part of the standard workflow and should only be executed by the CI/CD system.
+
+== Customization
+
+You can customize the different steps through the following environment variables:
+
+* `IMAGE_NAME` (Defaults to `helloworld`): name of the image built on the local container tool.
+* `IMAGE_DEPLOY_NAME` (Defaults to the same value as `IMAGE_NAME`): final name of the image to be pushed to a remote Docker registry service (ideally after the test harness success to avoid dangling and untested artefacts).
+* `DOCKERFILE` (Defaults to `Dockerfile`): the relative path to the Dockerfile to be built under the current ($(PWD)) context directory.
+* `IMAGE_ARCHIVE` (Defaults to `image.tar`): the path to the TAR archive continaining the docker image.
+* `CONTAINER_BIN` (Defaults to `img`): The container's command line tool to use for building and deploying the image.
+* The "metadata" informations passed as both labels and arguments to the build:
+** `GIT_COMMIT_REV` (Defaults to current commit's SHA): Git reference of the Docker image's code.
+** `GIT_SCM_URL` (Defaults to the URL of the current git's remote named "origin"): URL of the git SCM repository's remote of the Docker image's code.
+** `SCM_URI` (Defaults from `GIT_SCM_URL` value): HTTPS (enforced) URL of `GIT_SCM_URL`.
+** `BUILD_DATE` (Default to "now" when called): UTC timestamp formatted in `%Y-%m-%dT%H:%M:%S` when the command occurs.
+
+You can check the file `Makefile` if you want to get a better deep-dive understanding.
+
+
+== Contributing
+
+* Contributing to this documentation or tooling must follow the contribution rules of this repository (please check link:../../../../../README.adoc#Contributing[])
```

Comment:
The link here leads to the README for repo, which doesn't have much in the way of Contributing Guidelines.

Reply:
Really good point. Incoming PR to add a contribution guide to the repo (as it is.. lacking).

Reply:
The missing "contribution" doc is now tracked in https://issues.jenkins.io/browse/INFRA-2881 , and tagged as a "newbie issue" as it is a topic that can be taken by contributors with no prior Jenkins expertise.